### ISSUE TITLE
[3.5beta3, beta4] Unexpected reset when using laser mode

### ISSUE THREAD

**Issue by dc42** [dc42/RepRapFirmware#869]
### Duet Forum Discussion Thread

https://forum.duet3d.com/topic/32658/3-5-0-b3-reboot-in-laser-mode

### Which Duet products are you using?

- [X] Duet2-Wifi
- [ ] Duet2-Ethernet
- [ ] Duet Expansion Breakout Board
- [ ] Duex2
- [ ] Duex5
- [ ] Duet2-Maestro
- [ ] Maestro Dual Driver Expansion
- [ ] Duet3-6HC
- [ ] Duet3-3HC
- [ ] Duet3-1XD
- [ ] Duet3-1LC
- [ ] Duet3-Tool Distribution Board
- [ ] Duet3-Mini5+
- [ ] Duet3-Mini2+
- [ ] Raspberry Pi or other SBC
- [ ] SmartEffector
- [ ] Magnetic Filament Sensor
- [ ] Laser Filament Sensor
- [ ] PT100 Daughterboard
- [ ] Thermocouple Daughterboard
- [ ] PanelDue
- [ ] Other
- [ ] None

### Firmware Version

RRF 3.5.0-beta.3, 3.5.0-beta.4

### Duet Web Control Version

3.5.0-beta.4

### Are you using a Single Board Computer (RaspberryPi) with your Duet?

- [ ] Yes I use a SBC.
- [X] No I do not use a SBC.

### What happened?

Board resets when homing the printer after power up.

**Comment by dc42** [comment]
I haven't reproduced this issue but from the stack trace I believe it is caused by this code in GCodes::ReadMove:

```
	if (machineType == MachineType::laser)
	{
		ms.laserPwmOrIoBits.laserPwm = (ms.isCoordinated && ms.segmentsLeft < ms.laserPixelData.numPixels && !ms.doingArcMove)
										? ms.laserPixelData.pixelPwm[ms.laserPixelData.numPixels - ms.segmentsLeft]
											: (ms.isCoordinated && ms.laserPixelData.numPixels == 1)
											  ? ms.laserPixelData.pixelPwm[0]
												: 0;
	}
```
Variable ms.laserPixelData.numPixels was not initialised, which led to an out-of-bounds access in expression ```ms.laserPixelData.pixelPwm[ms.laserPixelData.numPixels - ms.segmentsLeft].```

Fixed in temporary 3.5.0-beta.4+ build.

**Comment by dc42** [comment]
Confirmed by the user as solved, see the forum thread.
